spec/features/navbar_changes_user_session_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe 'Navbar changes for user session' do
+  describe 'No user signed in' do
+    before do
+      visit 'welcome/index'
+    end
+
+    it 'displays link to new_user_registration' do
+      within("ul.navbar-right") do
+        expect( page ).to have_content "Sign Up"
+      end
+    end
+
+    it 'displays link to new_user_session' do
+      within("ul.navbar-right") do
+        expect( page ).to have_content "Sign In"
+      end
+    end
+  end
+
+  describe 'Signed in user' do
+
+    before do
+      @user = create(:user) # Confirmed user
+    end
+
+    it 'displays user first_name in navbar' do
+     sign_in_user
```

Comment:
`sign_in_user` is common to every `it` block here, so you could consider moving it into the `before` block.